improver/psychrometric_calculations/cloud_top_temperature.py

```diff
@@ -81,61 +82,62 @@ def _calculate_cct(self) -> ndarray:
         """
         p_coord = self.temperature.coord("pressure")
         t_data = [t.data for t in self.temperature.slices_over("pressure")]
```

Comment:
I tried this and ran both versions with the `memprofile` (Using one of the acceptance tests). No difference. Both gradually increase memory, probably because the net result is that all the lazy data become realized data in both cases, it just gets sliced up differently. I've asked Anna Booton if she knows how to discard data once used - might save us a huge amount in various places if we can.

Reply:
Tried again with full-size global data (18 members, whole globe). This takes 3.5 minutes to run and @bayliffe's suggestion reduces the peak memory size from 1320 MiB to 930 MiB.